.github/workflows/test-cygwin.yml

```diff
@@ -0,0 +1,65 @@
+name: Test Cygwin
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  build:
+    runs-on: windows-2019
+    name: Cygwin Python 3.8
+
+    steps:
+      - name: Checkout Pillow
+        uses: actions/checkout@v2
+
+      - name: Set up Cygwin
+        uses: egor-tensin/setup-cygwin@v3
+        with:
+          platform: x64
+          packages: >
+            libfreetype-devel libimagequant-devel libjpeg-devel
+            liblcms2-devel libopenjp2-devel libraqm-devel
+            libtiff-devel libwebp-devel libxcb-devel netpbm
+            ImageMagick jpeg xorg-server-extra xorg-server-common
+            xinit python38-setuptools python38-devel python38-tkinter
+            python38-wheel python38-pip python38-olefile
+            python38-numpy python38-pytest python38-sphinx
+            python38-packaging python38-cffi
+            libfribidi-devel zlib-devel libX11-xcb-devel tcl-tk-devel
+          install-dir: 'C:\tools\cygwin'
+
+      - name: Clean up path
+        uses: egor-tensin/cleanup-path@v2
+        with:
+          dirs: 'C:\tools\cygwin\bin;C:\tools\cygwin\lib\lapack'
+
+      - name: Build system information
+        run: C:\tools\cygwin\bin\python3.8 .github/workflows/system-info.py
+
+      - name: Ensure Python dependencies are installed
+        run: |
+          C:\tools\cygwin\bin\python3.8 -m pip install pip wheel setuptools
+          C:\tools\cygwin\bin\python3.8 -m pip install -r requirements.txt
+
+      - name: Build Pillow
+        run: |
+          C:\tools\cygwin\bin\python3.8 setup.py bdist_wheel
+
+      - name: Install Pillow
+        run: |
+          C:\tools\cygwin\bin\dash -c '/usr/bin/python3.8 -m pip install dist/*.whl'
+
+      - name: Check Pillow configuration
+        run: |
+          C:\tools\cygwin\bin\python3.8 selftest.py
+
+      - name: Test Pillow
+        run: |
+          C:\tools\cygwin\bin\xvfb-run -s '-screen 0 1024x768x24' .ci/test.sh
+
+      - name: After success
+        run: |
+          C:\tools\cygwin\bin\dash .ci/after_success.sh
+
+      - name: Upload coverage
+        run: |
+          C:\tools\cygwin\bin\bash -c 'bash <(curl -s https://codecov.io/bash) -F'
```

Comment:
The bash uploader is deprecated and will stop working on Feb 1, 2022:

https://docs.codecov.com/docs/about-the-codecov-bash-uploader

Maybe use https://docs.codecov.com/docs/codecov-uploader or https://github.com/codecov/codecov-action?

Reply:
Is the current version better?  The Windows and Linux codecov uploaders will also need to be updated, but that's a separate PR, I think.